identity-account/src/account/account.rs

```diff
@@ -193,58 +206,45 @@ impl Account {
   where
     U: Serialize + SetSignature,
   {
-    let snapshot: IdentitySnapshot = self.load_snapshot().await?;
-    let state: &IdentityState = snapshot.identity();
+    let state: &IdentityState = self.state();
 
-    let fragment: Fragment = Fragment::new(fragment);
-    let method: &TinyMethod = state.methods().fetch(fragment.name())?;
-    let location: &KeyLocation = method.location();
+    let method: &IotaVerificationMethod = state.document().resolve_method(fragment).ok_or(Error::MethodNotFound)?;
 
-    state.sign_data(self.did(), self.storage(), location, target).await?;
+    let location: KeyLocation = state.method_location(method.key_type(), fragment.to_owned())?;
```

Comment:
Good catch. I believe with the upcoming key refactor this should be addressed adequately, since the key location will be derived from the document consisting of fragment + a hash of the `MethodData`, which should only point methods with the same fragment and different DIDs to the same location, if the key material is the same, in which case that is the correct behavior.